src/vs/platform/shell/node/shellEnv.ts

```diff
@@ -137,6 +137,9 @@ async function doResolveUnixShellEnv(logService: ILogService, token: Cancellatio
 		} else if (name === 'nu') { // nushell requires ^ before quoted path to treat it as a command
 			command = `^'${process.execPath}' ${extraArgs} -p '"${mark}" + JSON.stringify(process.env) + "${mark}"'`;
 			shellArgs = ['-i', '-l', '-c'];
+		} else if (name === 'xonsh') { // #200374: xonsh expects escaping to avoid Python syntax
+			command = `$['${process.execPath}' ${extraArgs} -p '"${mark}" + JSON.stringify(process.env) + "${mark}"']`;
```

Comment:
My preference would be not to repeat the whole command, but since that's the pattern that was established by `nu`, I wanted to be minimally invasive and follow the patterns already in use.